src/cpu/aarch64/injectors/jit_uni_eltwise_injector.cpp

```diff
@@ -0,0 +1,2253 @@
+    h->eor(ZRegD(IDX(vmm_src)), ZRegD(IDX(vmm_src)), ZRegD(IDX(vmm_src)));
+    // set zeroes at those points which were < log(FLT_MIN)
+    blend_with_mask(vmm_aux2, vmm_src);
+
+    // compute polynomial
+    h->mov(ZRegD(IDX(vmm_src)), ZRegD(IDX(table_val(exp_pol, 4))));
+    h->fmad(vmm_src, p_all / T_m, vmm_aux1, ZRegS(IDX(table_val(exp_pol, 3))));
+    h->fmad(vmm_src, p_all / T_m, vmm_aux1, ZRegS(IDX(table_val(exp_pol, 2))));
+    h->fmad(vmm_src, p_all / T_m, vmm_aux1, ZRegS(IDX(table_val(exp_pol, 1))));
+    h->fmad(vmm_src, p_all / T_m, vmm_aux1, ZRegS(IDX(table_val(exp_pol, 0))));
+    h->fmad(vmm_src, p_all / T_m, vmm_aux1, ZRegS(IDX(table_val(one))));
+
+    // y = y * 2^n
+    h->fmul(vmm_src, vmm_src, vmm_aux2);
+    h->fmul(vmm_src, vmm_src, ZRegS(IDX(table_val(two))));
+}
+
+template <cpu_isa_t isa>
+void jit_uni_eltwise_injector_f32<isa>::relu_compute_vector_fwd(
+        const TRegS &vmm_src) {
+    h->not_(p_tmp0.b, h->P_ALL_ONE / T_z, PRegB(IDX(p_all)));
+    h->mov(ZRegD(IDX(vmm_aux1)), ZRegD(IDX(vmm_src)));
+    h->mov(vmm_aux1, p_tmp0 / T_m, 0);
+    compute_cmp_mask(vmm_src, table_val(zero), _cmp_gt_os);
+    h->fmul(vmm_src, vmm_src, ZRegS(IDX(table_val(alpha))));
+    blend_with_mask(vmm_src, vmm_aux1);
+}
+
+template <cpu_isa_t isa>
+void jit_uni_eltwise_injector_f32<isa>::relu_zero_ns_compute_vector_fwd(
+        const TRegS &vmm_src) {
+    h->mov(PRegB(IDX(p_tmp0)), h->P_ALL_ONE.b);
+    h->mov(ZRegD(IDX(z_tmp)), ZRegD(IDX(table_val(zero))));
+    h->fmaxnm(z_tmp, p_tmp0, vmm_src);
+    h->fmax(z_tmp, p_tmp0, vmm_src);
+    h->mov(ZRegD(IDX(vmm_src)), ZRegD(IDX(z_tmp)));
+}
+
+template <cpu_isa_t isa>
+void jit_uni_eltwise_injector_f32<isa>::elu_compute_vector_fwd(
+        const TRegS &vmm_src) {
+    // IMPORTANT: we use vmm_aux3 for the mask as exp_compute does not use it.
+    h->not_(p_tmp0.b, h->P_ALL_ONE / T_z, PRegB(IDX(p_all)));
+    h->mov(ZRegD(IDX(vmm_aux3)), ZRegD(IDX(vmm_src)));
+    h->mov(vmm_aux3, p_tmp0 / T_m, 0);
+
+    // compute exponent
+    exp_compute_vector_fwd(vmm_src);
+
+    // alpha * (exp(x) - 1)
+    h->fsub(vmm_src, vmm_src, ZRegS(IDX(table_val(one))));
+    h->fmul(vmm_src, vmm_src, ZRegS(IDX(table_val(alpha))));
+
+    // combine with mask
+    compute_cmp_mask(vmm_aux3, table_val(zero), _cmp_gt_os);
+    blend_with_mask(vmm_src, vmm_aux3);
+}
+
+template <cpu_isa_t isa>
+void jit_uni_eltwise_injector_f32<isa>::tanh_compute_vector_fwd(
+        const TRegS &vmm_src) {
+    // we add a check as the avx2 code cannot be used for avx
+    using namespace Xbyak_aarch64::util;
+    const int tanh_n_polynomials = 32;
+
+    // register mapping
+    // TODO: put sign on stack and alias zmm_table2 with vmm_sign to save a reg ?
+    TRegS vmm_dst = vmm_aux1, vmm_src_shift = vmm_aux1, vmm_coeff = vmm_aux1,
+          vmm_pol = vmm_aux2, vmm_indices = vmm_aux3,
+          vmm_src_original = vmm_aux4, vmm_sign = vmm_aux4;
+
+    auto vpermt2ps_aarch64 =
```

Comment:
I don't know if `vpermt2ps` makes sense in the aarch64 world :-)
Since it just does a table lookup of 64 entries in 2 zreg vectors with f32 entries, you could use `zreg_lut_64_f32` or `lut_zreg2_f32`.

Reply:
Thank you for the suggestion. I'll use `lut_zreg2_f32`.